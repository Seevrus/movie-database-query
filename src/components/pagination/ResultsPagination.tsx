import { Col, Container, Pagination, Row } from 'react-bootstrap';

const fillItems = (first: number, last: number) => {
  const items = [];
  for (let i = first; i <= last; i++) {
    items.push(i);
  }

  return items;
};

interface ResultsPaginationProps {
  activePage: number;
  setActivePage: React.Dispatch<React.SetStateAction<number>>;
  numberOfPages: number;
}

const ResultsPagination = ({
  activePage,
  setActivePage,
  numberOfPages,
}: ResultsPaginationProps) => {
  const firstNumber = activePage - 2 > 1 ? activePage - 2 : 1;
  const lastNumber = activePage + 2 <= numberOfPages ? activePage + 2 : numberOfPages;

  return (
    <Container fluid>
      <Row className="justify-content-md-center">
        <Col md="4">
          <Pagination>
            <Pagination.First onClick={() => setActivePage(1)} />
            <Pagination.Prev
              onClick={() => activePage - 1 > 0 && setActivePage(activePage - 1)}
            />
            {activePage > 2 && firstNumber > 1 && 
              <>
                <Pagination.Item onClick={() => setActivePage(1)}>{1}</Pagination.Item>
                <Pagination.Ellipsis />
              </>
            }

            {fillItems(firstNumber, lastNumber).map(
              (item) => (
                <Pagination.Item
                  key={item}
                  active={item === activePage}
                  onClick={() => setActivePage(item)}
                >
                  {item}
                </Pagination.Item>
              ),
            )}

            {activePage < numberOfPages - 1 && lastNumber < numberOfPages &&
              <>
                <Pagination.Ellipsis />
                <Pagination.Item 
                  onClick={() => setActivePage(numberOfPages)}
                >
                  {numberOfPages}
                </Pagination.Item>
              </>
            }
            <Pagination.Next
              onClick={() =>
                activePage + 1 <= numberOfPages && setActivePage(activePage + 1)
              }
            />
            <Pagination.Last onClick={() => setActivePage(numberOfPages)} />
          </Pagination>
        </Col>
      </Row>
    </Container>
  );
};

export default ResultsPagination;
