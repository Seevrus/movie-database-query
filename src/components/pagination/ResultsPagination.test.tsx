import renderer from 'react-test-renderer';
import ResultsPagination from './ResultsPagination';

describe('ResultsPagination', () => {
  test.each([1, 2, 3, 4, 5, 6])('renders correctly for page %i', (page) => {
    const component = renderer.create(
      <ResultsPagination activePage={page} numberOfPages={6} setActivePage={jest.fn()} />,
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
