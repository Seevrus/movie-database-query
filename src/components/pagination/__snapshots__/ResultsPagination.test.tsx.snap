// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsPagination renders correctly for page 1 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-md-4"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              «
            </span>
            <span
              className="visually-hidden"
            >
              First
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="visually-hidden"
            >
              Previous
            </span>
          </a>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            disabled={false}
            onClick={[Function]}
          >
            1
            <span
              className="visually-hidden"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            2
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            3
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              …
            </span>
            <span
              className="visually-hidden"
            >
              More
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            6
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              »
            </span>
            <span
              className="visually-hidden"
            >
              Last
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ResultsPagination renders correctly for page 2 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-md-4"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              «
            </span>
            <span
              className="visually-hidden"
            >
              First
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="visually-hidden"
            >
              Previous
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            1
          </a>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            disabled={false}
            onClick={[Function]}
          >
            2
            <span
              className="visually-hidden"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            3
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            4
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              …
            </span>
            <span
              className="visually-hidden"
            >
              More
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            6
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              »
            </span>
            <span
              className="visually-hidden"
            >
              Last
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ResultsPagination renders correctly for page 3 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-md-4"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              «
            </span>
            <span
              className="visually-hidden"
            >
              First
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="visually-hidden"
            >
              Previous
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            1
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            2
          </a>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            disabled={false}
            onClick={[Function]}
          >
            3
            <span
              className="visually-hidden"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            4
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            5
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              …
            </span>
            <span
              className="visually-hidden"
            >
              More
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            6
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              »
            </span>
            <span
              className="visually-hidden"
            >
              Last
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ResultsPagination renders correctly for page 4 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-md-4"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              «
            </span>
            <span
              className="visually-hidden"
            >
              First
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="visually-hidden"
            >
              Previous
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            1
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              …
            </span>
            <span
              className="visually-hidden"
            >
              More
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            2
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            3
          </a>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            disabled={false}
            onClick={[Function]}
          >
            4
            <span
              className="visually-hidden"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            5
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            6
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              »
            </span>
            <span
              className="visually-hidden"
            >
              Last
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ResultsPagination renders correctly for page 5 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-md-4"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              «
            </span>
            <span
              className="visually-hidden"
            >
              First
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="visually-hidden"
            >
              Previous
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            1
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              …
            </span>
            <span
              className="visually-hidden"
            >
              More
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            3
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            4
          </a>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            disabled={false}
            onClick={[Function]}
          >
            5
            <span
              className="visually-hidden"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            6
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              »
            </span>
            <span
              className="visually-hidden"
            >
              Last
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`ResultsPagination renders correctly for page 6 1`] = `
<div
  className="container-fluid"
>
  <div
    className="justify-content-md-center row"
  >
    <div
      className="col-md-4"
    >
      <ul
        className="pagination"
      >
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              «
            </span>
            <span
              className="visually-hidden"
            >
              First
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ‹
            </span>
            <span
              className="visually-hidden"
            >
              Previous
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            1
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              …
            </span>
            <span
              className="visually-hidden"
            >
              More
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            4
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            5
          </a>
        </li>
        <li
          className="page-item active"
        >
          <span
            className="page-link"
            disabled={false}
            onClick={[Function]}
          >
            6
            <span
              className="visually-hidden"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              ›
            </span>
            <span
              className="visually-hidden"
            >
              Next
            </span>
          </a>
        </li>
        <li
          className="page-item"
        >
          <a
            className="page-link"
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              aria-hidden="true"
            >
              »
            </span>
            <span
              className="visually-hidden"
            >
              Last
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
